org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/properties/RulesConfigurationPart.java

```diff
@@ -208,19 +211,32 @@ public int compare(Viewer viewer, Object e1, Object e2) {
     tree.getViewer().setInput(ruleDetailsWrappersByLanguage.keySet().toArray(new Language[ruleDetailsWrappersByLanguage.size()]));
   }
 
+  /** Show GetEffectiveRuleDetailsResponse as well??? */
   private void refreshUiForRuleSelection(Object selectedNode) {
     paramPanel.dispose();
     if (selectedNode instanceof RuleDetailsWrapper) {
       var wrapper = (RuleDetailsWrapper) selectedNode;
-      ruleBrowser.updateRule(wrapper.ruleDetails);
-      if (wrapper.ruleDetails.paramDetails().isEmpty()) {
-        paramPanel = emptyRuleParam();
-      } else {
-        paramPanel = new RuleParameterPanel(paramPanelParent, SWT.NONE, wrapper.ruleDetails, wrapper.ruleConfig);
-        paramPanel.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+      var ruleKey = wrapper.ruleDetails.getKey();
+
+      try {
+        ruleBrowser.updateRule(SonarLintBackendService.get()
+          .getBackend()
+          .getRulesService()
+          .getStandaloneRuleDetails(new GetStandaloneRuleDescriptionParams(ruleKey))
+          .get());
```

Comment:
This is probably interesting to do this in a background job, in order to not freeze the UI